platform/viewer/src/components/ViewportGrid/ViewportGrid.js

```
@@ -0,0 +1,186 @@
+          {...viewportProps}
+          // We shouldn't need this?
+          // Used in `ConnectedCornerstoneViewport`
+          viewportIndex={viewportIndex}
+          setViewportActive={() => {
+            if (activeViewportIndex !== viewportIndex) {
+              setViewportActive(viewportIndex);
+            }
+          }}
+        />
+        {childrenWithProps}
+      </ViewportPane>
+    );
+  });
+
+  const rowSize = 100 / numRows;
+  const colSize = 100 / numColumns;
+
+  // http://grid.malven.co/
+  return (
+    <div
+      style={{
+        display: 'grid',
+        gridTemplateRows: `repeat(${numRows}, ${rowSize}%)`,
+        gridTemplateColumns: `repeat(${numColumns}, ${colSize}%)`,
+        height: '100%',
+        width: '100%',
+      }}
+    >
+      {magic}
+    </div>
+  );
+};
+
+const noop = () => {};
+
+ViewportGrid.propTypes = {
+  activeViewportIndex: PropTypes.number.isRequired,
+  numRows: PropTypes.number.isRequired,
+  numColumns: PropTypes.number.isRequired,
+  viewportPanes: PropTypes.arrayOf(
+    PropTypes.shape({
+      studyInstanceUid: PropTypes.string,
+      displaySetInstanceUid: PropTypes.string,
+      plugin: PropTypes.string,
+    })
+  ).isRequired,
+  availablePlugins: PropTypes.object.isRequired,
+  setViewportData: PropTypes.func,
+  setViewportActive: PropTypes.func, // Connected
+  studies: PropTypes.array,
+  children: PropTypes.node,
+};
+
+ViewportGrid.defaultProps = {
+  activeViewportIndex: 0,
+  availablePlugins: {
+    DefaultViewport,
+  },
+  setViewportActive: noop,
+  defaultPlugin: 'DefaultViewport',
+};
+
+/**
+ *
+ *
+ * @param {*} studies
+ * @param {string} studyInstanceUid
+ * @param {string} displaySetInstanceUid
+ * @returns
+ */
+function _getDisplaySetStackFromStudies(
+  studies,
+  studyInstanceUid,
+  displaySetInstanceUid
+) {
+  if (!studyInstanceUid || !displaySetInstanceUid) {
+    return undefined;
```

`return;` is identical, though maybe we should just throw an error

---

There is a fair amount here that we need to talk about. I have a running list of questions around loading data and passing them in an appropriate/consistent way to viewports.